### PR TITLE
Adds a puppet module for Duo MFA and the ability to configure it from the Terraform deployment

### PR DESCRIPTION
**Pull request by netscruff**
This PR adds a Puppet module forked from the Duo Puppet module which was abandoned by Duo but picked up by Indiana University. The Duo configuration can be defined by creating a yaml files with the Duo api keys and included in user data when deploying a new cluster. By default no MFA deployment is done, the class is not added to node types but you can configure MFA on a type of node (login, mgmt, node) by adding the appropriate user data configuration.

profile::mfa::duo::login: true
profile::mfa::duo::mgmt: true
profile::mfa::duo::node: true

### CODE REVIEW

**Comment by cmd-ntrf** [comment]
I suggest the classes to be renamed:
- `profile::mfa::login`
- `profile::mfa::mgmt`
- `profile::mfa::node`

These classes would have a `Enum['none', 'duo'] provider` argument, so if we add more MFA options eventually, it will mostly be  matter of adding a value to the enum and the appropriate logic for that provider in the classes. The default value in the hieradata for the provider would be `none`.

File: manifests/site.pp [hunk 1]
```diff
@@ -23,6 +23,7 @@
   include profile::reverse_proxy
   include profile::nfs::client
   include profile::freeipa::client
+  include profile::mfa::duo::login
```
Comment (cmd-ntrf) [inline manifests/site.pp:26]: This needs to be updated to reflect the new class name.

File: manifests/site.pp [hunk 2]
```diff
@@ -40,6 +41,7 @@
   include profile::freeipa::guest_accounts
   include profile::slurm::accounting
   include profile::workshop::mgmt
+  include profile::mfa::duo::mgmt
```
Comment (cmd-ntrf) [inline manifests/site.pp:44]: This needs to be updated to reflect the new class name.

File: manifests/site.pp [hunk 3]
```diff
@@ -49,6 +51,7 @@
   include profile::rsyslog::client
   include profile::freeipa::client
   include profile::metrics::exporter
+  include profile::mfa::duo::mgmt
```
Comment (cmd-ntrf) [inline manifests/site.pp:54]: This needs to be updated to reflect the new class name.

File: manifests/site.pp [hunk 4]
```diff
@@ -64,4 +67,6 @@
   include profile::nfs::client
   include profile::slurm::node
   include profile::freeipa::client
+  include profile::mfa::duo::node
```
Comment (cmd-ntrf) [inline manifests/site.pp:70]: This needs to be updated to reflect the new class name.

File: README.md [hunk 5]
```diff
@@ -92,3 +92,21 @@ variables for each profile.
 | --------------------------------- | :----- | :------------------------------------------------------------------------------ | ------------------------ |
 | `profile::workshop::userzip_url`  | String | URL pointing to a zip that needs to be extracted in each guest account's home   | `''`                     |
 | `profile::workshop::userzip_path` | String | Path on the nfs server where to save the userzip archive                        | `'/project/userzip.zip'` |
+
+## profile::mfa
+
+| Variable                              | Type           | Description                                                                 | Default  |
+| ------------------------------------- | :------------- | :-------------------------------------------------------------------------- | -------- |
+| `profile::mfa::duo:mgmt`              | Boolean        | Enable MFA on mgmt nodes                                                    | false    |
```
Comment (cmd-ntrf) [inline README.md:100]: This needs to be updated to reflect the new class name.

File: README.md [hunk 6]
```diff
@@ -92,3 +92,21 @@ variables for each profile.
...
+| `profile::mfa::duo:login`             | Boolean        | Enable MFA on login nodes                                                   | false    |
```
Comment (cmd-ntrf) [inline README.md:101]: This needs to be updated to reflect the new class name.

File: README.md [hunk 7]
```diff
@@ -92,3 +92,21 @@ variables for each profile.
...
+| `profile::mfa::duo:node`              | Boolean        | Enable MFA on compute nodes                                                 | false    |
```
Comment (cmd-ntrf) [inline README.md:102]: This needs to be updated to reflect the new class name.

**Comment by cmd-ntrf** [comment]
GTM

**Comment by netscruff** [comment]
Just tested the last changes to common.yaml on a deployed cluster.

**Pull request merged**